helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -392,17 +450,18 @@ public void invoke(NotificationContext changeContext) throws Exception {
         logger.warn("Callback handler received event in wrong order. Listener: " + _listener
```

Comment:
My assumption is that you have a new type to by pass when there is a callback type wrong order. But here you may still have the problem that current type is FINALIZE and coming with CALLBACK. And your type PERODIC_REFRESH will also be recognized as a wrong order type.

I would suggest rethink the value of adding the new type. What is the purpose to have a new type.

Reply:
I agree. I have updated the code, please take a look.